docs/essentials/configuring-pre-flight-checks.md

```diff
@@ -0,0 +1,130 @@
+    of potential ramifications from ill-defined pre-flight checks, including failure of all
+    your CI/CD pipelines in Semaphore 2.0.
+
+As a security feature, pre-flight checks can be modified only by people who can:
+
+- **manage organization settings** - organization pre-flight checks,
+- **manage project settings** - project pre-flight checks.
+
+## Configuring pre-flight checks
+
+### Organization pre-flight checks
+
+Follow the steps below to configure pre-flight checks for the whole organization:
+
+1. Open the **Settings** from the **Organization menu** in the right part of the page header.
+
+2. Click **Pre-flight checks** on the left side of the settings view.
+
+3. Type **Commands** to execute before each pipeline. 
+
+4. Choose **Secrets** you want to use in **Commands**. 
+
+5. Provide **Agent configuration** for pipeline initialization job. 
+
+    You can choose between different machine types:
+
+    - **Linux Based Virtual Machines** - hosted by Semaphore
+    - **Mac Based Virtual Machines** - hosted by Semaphore
+    - **Self-Hosted Machines** - hosted by the customer (if applicable)
+
+    For machines hosted by Semaphore, choose a proper **Machine type** and **OS image** 
+    of the agent. 
+    
+    For self-hosted machines choose a **Machine type** that matches your
+    self-hosted agent type.
+
+6. Click **Save changes** button.
+
+To remove pre-flight checks for the whole organization, click the red button
+**Delete pre-flight checks**, which should be visible if you have pre-flight checks
+configured.
+
+### Project pre-flight checks
+
+Configuring project pre-flight checks requires you to follow these steps:
+
+1. Open the project page and go to the **Settings** tab.
+
+2. Click **Pre-flight checks** on the left side of the settings view.
+
+3. Type **Commands** to execute before each pipeline.     
+
+4. Choose **Secrets** you want to use in **Commands**. 
+
+5. Click **Save changes** button.
+
+To remove pre-flight checks for the project, just click **Delete pre-flight checks**
+button, which should be visible if you have pre-flight checks configured.
+
+## Examples of pre-flight checks configuration
+
+### Limiting using secrets in your organization
+
+In the initialization job, the following information is available as environment variables:
+
+- [the project name](/ci-cd-environment/environment-variables/#semaphore_project_name) for which the pipeline started
+- [repository branch](/ci-cd-environment/environment-variables/#semaphore_git_branch) for which the pipeline started
+- YAML file describing the running pipeline (`SEMAPHORE_YAML_FILE_PATH`)
+
+Based on that, you can define commands in pre-flight checks ensuring that
+a particular pipeline can use certain secrets. You can use this [snippet](https://gist.githubusercontent.com/shiroyasha/49b57770371f4684a8a12bcec0047d7a/raw/b35e51953069dc1b21c62a41b8717aa5fb4bcf88/check-secret)
```

Comment:
We need a different location for this snippet, [renderedtext/snippets](https://github.com/renderedtext/snippets) comes to mind, but maybe @AleksandarCole or @shiroyasha have a better idea where we can put it.